db/table_properties_collector.h

```diff
@@ -108,4 +110,60 @@ class UserKeyTablePropertiesCollectorFactory
   std::shared_ptr<TablePropertiesCollectorFactory> user_collector_factory_;
 };
 
+class TimestampTablePropertiesCollector : public IntTblPropCollector {
+ public:
+  explicit TimestampTablePropertiesCollector(const Comparator* cmp)
+      : cmp_(cmp) {}
+
+  virtual ~TimestampTablePropertiesCollector() {}
+
+  Status InternalAdd(const Slice& key, const Slice& /* value */,
+                     uint64_t /* file_size */) override {
+    if (!cmp_) {
+      return Status::OK();
+    }
+    auto timestampInKey =
+        ExtractTimestampFromUserKey(key, cmp_->timestamp_size());
+    if (timestampInKey == kDisableUserTimestamp) {
+      return Status::OK();
+    }
+    if (max_timestamp == kDisableUserTimestamp ||
+        cmp_->CompareTimestamp(timestampInKey, max_timestamp)) {
+      max_timestamp = timestampInKey.ToString();
+    }
+    if (min_timestamp == kDisableUserTimestamp ||
+        cmp_->CompareTimestamp(min_timestamp, timestampInKey)) {
+      min_timestamp = timestampInKey.ToString();
+    }
+    return Status::OK();
+  }
+
+  void BlockAdd(uint64_t /* block_raw_bytes */,
+                uint64_t /* block_compressed_bytes_fast */,
+                uint64_t /* block_compressed_bytes_slow */) override {
+    // left blank
+    return;
+  }
+
+  Status Finish(UserCollectedProperties* properties) override {
+    properties->insert({"min_timestamp", min_timestamp});
+    properties->insert({"max_timestamp", max_timestamp});
+    return Status::OK();
+  }
+
+  const char* Name() const override {
+    return "TimestampTablePropertiesCollector";
+  }
+
+  UserCollectedProperties GetReadableProperties() const override {
+    // left blank
```

Comment:
What does this comment mean? No real implementation for this override method?

Reply:
overrides pure virtual

Reply:
Yeah, this is for overrides pure virtual fuction.  Already removed this comment.